nixos/tests/installed-tests/default.nix

```diff
@@ -48,7 +50,15 @@ let
               testConfig
             ] ++ optional withX11 ../common/x11.nix;
 
-            environment.systemPackages = with pkgs; [ gnome-desktop-testing ];
+            environment.systemPackages = [
+              (pkgs.writeShellScriptBin "nixos-installed-tests-runner" ''
+                runner() {
+                  ${pkgs.gnome-desktop-testing}/bin/gnome-desktop-testing-runner ${escapeShellArgs testRunnerFlags} -d '${tested.installedTests}/share' "$@"
+                }
+
+                runner ${optionalString (excludeTestRegex != null) "$(runner --list | cut -d ' ' -f 1 | grep -v ${escapeShellArg excludeTestRegex})"}
```

Comment:
I may try to add `--exclude` flag [upstream](https://gitlab.gnome.org/GNOME/gnome-desktop-testing) when I have some bandwidth. I recently [did the same](https://github.com/mesonbuild/meson/pull/11502) for Meson.

Reply:
That is definitely the best way to go, I'm not familiar with the codebase enough to touch it myself :(